caffe2/operators/layer_norm_op.cc

```diff
@@ -196,12 +196,32 @@ void layer_norm_c10(c10::Stack* stack, c10::KernelCache* cache_) { // TODO Pass
   c10::ArrayRef<c10::IValue> inputs = torch::jit::peekSlice(*stack, 0, 3, 6);
   c10::ArrayRef<c10::IValue> outputs = torch::jit::peekSlice(*stack, 3, 3, 6);
 
+
   caffe2::Tensor X{c10::C10Tensor(inputs[0].toTensor())};
   int64_t axis = inputs[1].toInt();
   float epsilon = inputs[2].toDouble();
-  caffe2::Tensor Y{c10::C10Tensor(outputs[0].toTensor())};
-  caffe2::Tensor mean{c10::C10Tensor(outputs[1].toTensor())};
-  caffe2::Tensor sig{c10::C10Tensor(outputs[2].toTensor())};
+
+  auto device = X.GetDevice();
+
+  caffe2::Tensor Y, mean, sig;
+  if (outputs[0].isTensor()) {
+    Y = caffe2::Tensor(c10::C10Tensor(std::move(outputs[0]).toTensor()));
+  }
+  if (outputs[1].isTensor()) {
+    mean = caffe2::Tensor(c10::C10Tensor(std::move(outputs[1]).toTensor()));
+  }
+  if (outputs[2].isTensor()) {
+    sig = caffe2::Tensor(c10::C10Tensor(std::move(outputs[2]).toTensor()));
+  }
+  if (!Y.defined()) {
```

Comment:
can't it be just in `else` branch above? and also it almost asks for a helper function

Reply:
https://github.com/pytorch/pytorch/pull/16548/files implements this in a more generic way, i.e. one implementation for all ops. In this PR here, I only have to add these checks because the outputs can be an empty IValue as mentioned above.